### ISSUE TITLE
How does one evaluate predict() during training?

### ISSUE THREAD

**Issue by rahkstar** [hill-a/stable-baselines#180]
Hi,

Apologies if this question doesn't below here. I've noticed while training mountain car with a dqn using the sample scripts the agent while training never converges to produce the same set of actions to reach the hilltop **every episode**. When I run the enjoy_mountain_car script it solves the environment on **every episode**. Is it possible to evaluate the true performance of the training agent during learn() without running an enjoy_mountain_car-like script and recording performance at every kth timestep? 

n.b. I'm now to RL

code below:
class CustomPolicy(FeedForwardPolicy):
    def __init__(self, *args, **kwargs):
        super(CustomPolicy, self).__init__(*args, **kwargs,
                                           layers=[64, 64],
                                           layer_norm=True,
                                           feature_extraction="mlp")

env = gym.make("MountainCar-v0")

model = DQN(
    policy=CustomPolicy,
    env=env,
    verbose=1,
    learning_rate=1e-3,
    buffer_size=50000,
    exploration_fraction=0.1,
    exploration_final_eps=0.1,
    param_noise=True,
    prioritized_replay=True,
    action_error_std=0
)
model.learn(total_timesteps=100000)

print("Saving model to mountaincar_model.pkl")
model.save("mountaincar_model.pkl")

while True:
    obs, done = env.reset(), False
    episode_rew = 0
    while not done:
        if not False:
            env.render()
        # Epsilon-greedy
        if np.random.random() < 0.02:
            action = env.action_space.sample()
        else:
            action, _ = model.predict(obs, deterministic=True)
        obs, rew, done, _ = env.step(action)
        episode_rew += rew
    print("Episode reward", episode_rew)
    # No render is only used for automatic testing

Thanks!

**Comment by araffin** [comment]
Hello,
I'm not sure to understand what you want to do exactly. But if you want to test you agent every n steps, you can do something like:

```python
from stable_baselines import DQN

model = DQN('MlpPolicy', 'MountainCar-v0', verbose=1)

# train your model for 1000 timesteps
model.learn(1000)

# here you can test it using model.predict()
...

# continue training
model.learn(1000)

```

**Comment by rahkstar** [comment]
much appreciated I will give it a try. 

Best
